### PR TITLE
fonts: add module

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: modules/default.nix [hunk 1]
```diff
@@ -16,6 +16,7 @@ let
     ./misc/gtk.nix
     ./misc/news.nix
     ./misc/pam.nix
+    ./config/fonts.nix
```
Comment (rycee) [inline modules/default.nix]: I think the `fonts.nix` file could go into the `misc` directory rather than a new `config` directory.

**Comment by rycee** [comment]
Thanks for the PR! Thus far I've simply put font packages into `home.packages`. Do you know of any benefit of using this method over using `home.packages`? In principle I like the idea of keeping fonts more organized but I'm worried that it doesn't actually add anything beyond what's possible already…

**Comment by Mic92** [comment]
Oh, are they picked up from their? I have actually not tried that and just copied the behavior from nixpkgs.

**Comment by uvNikita** [comment]
I also just assumed that adding fonts to the packages list will not work and managed all fonts in the system configuration :)

**Comment by Mic92** [comment]
So this module would be probably only useful for non-nixos system, I suppose.

**Comment by rycee** [comment]
That is a very good argument! Although I personally use NixOS everywhere, a goal of mine is for Home Manager to be useful for people using other distros.

I guess an alternative may be to have a boolean option `fonts.linkFontsDir` (or whatever) that on `true` causes
```
home.file.".local/share/fonts".source = "${config.home.path}/share/fonts";
```
to happen (`config.home.path` is the `pkgs.buildEnv` of all packages in `home.packages`). Then a non-NixOS user could also install fonts through `home.packages`…

I don't know which solution is most elegant but @Mic92 convinced me that it would be nice to have a way to get the fonts into `~/.local/share`. Do you have any preference?

**Comment by Mic92** [comment]
@rycee your approach sounds better.

**Pull request closed without merging**